File: Ontology/Annotation.cs
namespace Ontology
{
    public class Annotation:
        Annotated,
        IAnnotation
    {
        private readonly IAnnotationProperty _property;
        private readonly object              _value;

        public Annotation(
            IAnnotationProperty property,
            object              value
            ) : base()
        {
            _property = property;
            _value    = value;
        }

        IAnnotationProperty IAnnotation.Property => _property;

        object IAnnotation.Value => _value;
    }
}

File: Web/ClientApp/src/app/Ontology/Annotation.ts
import { Annotated } from "./Annotated";
import { IAnnotation } from "./IAnnotation";
import { IAnnotationProperty } from "./IAnnotationProperty";

export class Annotation
    extends Annotated
    implements IAnnotation
{
    constructor(
        public Property: IAnnotationProperty,
        public Value   : object
        )
    {
        super();
    }
}
